File: swiftui/Sources/ArenaTokens/ArenaMotion.swift
import CoreGraphics
import Foundation

public enum ArenaMotion {
    public static func working(_ loop: TimeInterval, reduced: Bool, slowed: TimeInterval = ArenaTokens.loopReduced) -> TimeInterval { reduced ? slowed : loop }

    public static func decorative(_ reduced: Bool) -> Bool { !reduced }

    public static func travel(_ distance: CGFloat, reduced: Bool) -> CGFloat { reduced ? ArenaTokens.sp0 : distance }
}

File: compose/src/main/kotlin/org/dravensoft/arena/tokens/ArenaMotion.kt
package org.dravensoft.arena.tokens

import androidx.compose.ui.unit.Dp

public object ArenaMotion {
    public fun working(loop: Int, reduced: Boolean, slowed: Int = ArenaTokens.loopReduced): Int = if (reduced) slowed else loop

    public fun decorative(reduced: Boolean): Boolean = !reduced

    public fun travel(distance: Dp, reduced: Boolean): Dp = if (reduced) ArenaTokens.sp0 else distance
}
